src/main/java/seedu/address/model/tag/TagType.java

```diff
@@ -0,0 +1,41 @@
+package seedu.address.model.tag;
+
+/**
+ * Enum class for type of tags.
+ */
+public enum TagType {
+    FRIEND("Friend"),
+    CLOSE_FRIEND("Close Friend"),
+    EMERGENCY("Emergency");
+
+    private final String displayName;
+
+    TagType(String displayName) {
+        this.displayName = displayName;
+    }
+
+    /**
+     * Method to return the string of tag display name.
+     *
+     * @return String of display name.
+     */
+    public String getDisplayName() {
+        return displayName;
+    }
+
+    /**
+     * Method to transform a string to a Tagtype.
+     *
+     * @param name name of the String.
+     * @return Tagtype of the name passed in.
+     */
+    public static TagType fromString(String name) {
```

Comment:
i think it's better to include "throws IlegalArgumentException" in the function header

Reply:
Hi Joel, thanks for pointing out this issue, I have edited such that it includes the throws execption.